salt/cloud/clouds/ec2.py

```diff
@@ -1178,7 +1179,7 @@ def securitygroupid(vm_):
                 log.debug('AWS SecurityGroup ID of {0} is {1}'.format(
                     sg['groupName'], sg['groupId'])
                 )
-                securitygroupid_set.add(sg['groupId'])
+                securitygroupid_set = securitygroupid_set.add(sg['groupId'])
```

Comment:
This will nullify ``securitygroupid_set``, since ``set.add()`` returns ``None``.